### PR TITLE
[CALCITE-3302] Implement CLASSIFIER function for MATCH_RECOGNIZE

### PR DESCRIPTION
**Pull request by JulianFeinauer**
This implements the CLASSIFIER command for MATCH_RECOGNIZE.

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/sql/parser/parserextensiontesting/ExtensionSqlParserTest.java [hunk 1]
```diff
@@ -32,7 +32,8 @@
 public class ExtensionSqlParserTest extends SqlParserTest {
 
   @Override protected SqlParserImplFactory parserImplFactory() {
-    return ExtensionSqlParserImpl.FACTORY;
+    // return ExtensionSqlParserImpl.FACTORY;
+    return null;
```
Comment (danny0405) [inline core/src/test/java/org/apache/calcite/sql/parser/parserextensiontesting/ExtensionSqlParserTest.java]: Why we return a null factory ?

Comment (JulianFeinauer) [reply core/src/test/java/org/apache/calcite/sql/parser/parserextensiontesting/ExtensionSqlParserTest.java]: @danny0405 thanks for the comment, you are absoluetly right. I comment that usually out (as it has sometimes compilation issues with maven) but should not have commited that. Is fixed now.

**Comment by julianhyde** [comment]
Sorry to be pedantic. Description should be "[CALCITE-3302] Implement CLASSIFIER function for MATCH_RECOGNIZE", not "CALCITE-3302 Implemented CLASSIFIER function for MATCH_RECOGNIZE."

**Comment by JulianFeinauer** [comment]
Agree, sorry for missing that, I changed the title accordingly.

**Comment by JulianFeinauer** [comment]
I'm closing this PR now as I already started to refactor it and reimplement it together with other functions like LAST. Nonetheless, thanks for your comments!

**Pull request closed without merging**